### ISSUE TITLE
OS X Finder reports 0 bytes available

### ISSUE THREAD

**Issue by spinkb** [EtiennePerot/fuse-jna#46]
Just trying out this project, ran directly from my eclipse.  Things work OK for my needs so far, except the OS X Finder reports 0 bytes available.  I was testing with the example memoryFS.

In terminal I can still copy items via cp, but not with the OS X Finder because it believes the server is full and won't make any attempts.  What can I do to troubleshoot for you?  Do you want access to an OS X machine with eclipse?  I know you mentioned many items before you didn't have OS X.

Other Fuse related apps I have used aren't reporting 0 bytes.  So its something specific to your implementation I think.

"df" also reports the volume is 100% utilized.

OS X 10.10.1.

Thanks,
Ben

**Comment by EtiennePerot** [comment]
`MemoryFS` is a minimal example filesystem that doesn't implement all the functions a real filesystem needs to (It doesn't have symlinks, it doesn't have locking, it doesn't have extended attributes, it doesn't even have regular attributes, it doesn't have device files...). It also doesn't implement `statfs`, which is equivalent to returning "the filesytem is 0 bytes large, 0 of them are used, and 0 of them are free". If you implement `statfs` in your filesystem, you should observe that the volume will report whatever numbers you make that function return.

If the Finder doesn't let you copy files because the filesystem is too full, that's kind of a problem for demo purposes though. Can you confirm that you can successfully copy files with `cp`? If the Finder really has that behavior, it may be worth implementing `statfs` in `MemoryFS` so that potential Mac OS X library users aren't turned off by this.

**Comment by spinkb** [comment]
Yes, I copied files with cp without issue. So having that may help then.  I will go ahead and implement my own filesystem for testing and see if I get Ok results then too.

**Comment by spinkb** [comment]
So I've implemented a full filesystem now to a remote http server.  Everything works.  I event added the statfs to report its a 1TB drive so the finder doesn't restrict any file copies.

So Thanks, that helped.  Only one odd kernel panic...which surprised me for FUSE.  But things are working OK so far, now a lot more app testing needs to be done.

--Ben

**Comment by EtiennePerot** [comment]
Keeping this bug open as a reminder to consider implementing `statfs` in the sample filesystems.
